interfaces/builtin/utils.go

```diff
@@ -155,7 +155,7 @@ func getDesktopFileRules(s *snap.Info) string {
 	denyRules := "# Explicitly deny access to other snap's desktop files\n"
 	desktopFiles, err := desktopFilesFromInstalledSnap(s)
 	if err != nil {
-		logger.Noticef("error: %v", err)
+		logger.Noticef("internal error: failed to collect desktop files from snap %q: %v", s.InstanceName(), err)
```

Comment:
I'm not sure this one and the one below are internal errors though, do we check desktop file names somewhere else?

Reply:
Good point, the snap mount could have been unmounted due to external reasoning which would cause failure. Updated.

Regarding the ones below:
- `internal error: invalid desktop file name %q found in snap %q which should have been validated earlier`
  - My reasoning here was that if we got so far and also got a bad desktop file name then snapd have been tricked somehow by the snap either when validating desktop file ids or when sanitizing the desktop file names during install/refresh.
- `internal error: failed to generate deny rules for snap %q: %v`
  - My reasoning here was that there is a case we didn't account for that breaks deny rules generation